src/frame.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+var STARTPINS = 10;
+
+function Frame(frameNum) {
+  this.balls = [];
+  this.complete = false;
+  this.frameNumber = frameNum;
+  this.pins = STARTPINS;
+}
+
+Frame.prototype.addBall = function (score) {
+  if(this.complete === false){
+    this.balls.push(score);
+    this._checkPins(score);
+    this._checkComplete();
+  }
+  else{
+    throw 'Cannot add ball to a frame that is already complete'
+  }
+}
+
+Frame.prototype.getBalls = function () {
+  return this.balls;
+};
+
+Frame.prototype.getFrameNumber = function(){
+  return this.frameNumber;
+}
+
+Frame.prototype.isComplete = function () {
+  return this.complete;
+};
+
+Frame.prototype.getFrameScore = function (nextBallOne = 0, nextBallTwo = 0) {
+  var score = this._calculateScore();
+  if(this.frameNumber < MAXFRAMES){
+    if(this.balls[0] === 10) { score = score + nextBallOne + nextBallTwo; }
+    if((this.balls[0] + this.balls[1]) > 9) { score = score + nextBallOne; }
+  }
+  return score;
+};
+
+Frame.prototype._getPins = function () {
+  return this.pins;
+};
+
+Frame.prototype._checkPins = function(score){
+  this.pins -= score;
+  if(this.balls.length === 1 && this.balls[0] === 10){
```

Comment:
Are there ways to increase the readability of expressions like `this.balls.length` and `this.balls[0]`?